core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -245,15 +245,24 @@ private[deploy] class Master(
       logError("Leadership has been revoked -- master shutting down.")
       System.exit(0)
 
-    case WorkerDecommission(id, workerRef) =>
-      logInfo("Recording worker %s decommissioning".format(id))
+    case WorkerDecommissioning(id, workerRef) =>
       if (state == RecoveryState.STANDBY) {
         workerRef.send(MasterInStandby)
       } else {
         // We use foreach since get gives us an option and we can skip the failures.
-        idToWorker.get(id).foreach(decommissionWorker)
+        idToWorker.get(id).foreach(w => decommissionWorker(w))
       }
 
+    case DecommissionWorkers(ids) =>
+      ids.foreach ( id =>
```

Comment:
shall we skip it for `if (state == RecoveryState.STANDBY)` ?

Reply:
or it will never happen because the request comes from master web UI?

Reply:
It has been checked by the caller when handling `DecommissionWorkersOnHosts`.

Reply:
then probably it's better to add an `assert(state != RecoveryState.STANDBY)` here

Reply:
ok, make sense.

Reply:
ah i see